src/_pytest/fixtures.py

```diff
@@ -1281,8 +1281,12 @@ def parsefactories(self, node_or_obj, nodeid=NOTSET, unittest=False):
                 marker = defaultfuncargprefixmarker
                 from _pytest import deprecated
 
-                self.config.warn(
-                    "C1", deprecated.FUNCARG_PREFIX.format(name=name), nodeid=nodeid
+                filename, lineno = getfslineno(obj)
+                warnings.warn_explicit(
+                    deprecated.FUNCARG_PREFIX.format(name=name),
```

Comment:
for internal use - we should **always** have the warning as instace of a warning type - we hould never deal a separate category

Reply:
Not sure what's your suggestion here @RonnyPfannschmidt... we are calling `warn_explicit`, which requires separate message and category parameters.

Reply:
indeed, thats unfortunate, i'd like to have a wrapper, and consistently use warnings as Warning instances

Reply:
I updated all `warn_explict` calls to use warning instances and pass `category=None`, at least.